Implements a StyleProcessor class to allow access to a list of set styles for a widget in String format.

neilcabanilla opened the pull request:
This PR allows us to fix issues with the CSS Spy in Eclipse PDE, detailed in [Issue #611](https://github.com/eclipse-pde/eclipse.pde/issues/611) Completed with @WillPeltz

**Changes**
-StyleProcessor class
Parses styles that can be set based on context, checks relevant styles against integer bitmask to see which are actually set, and converts them into String list

-getStyles method in Text and Button classes
Returns a list of styles currently set for a given widget in String format, allowing for style inspection at runtime

-Static function using instance of StyleProcessor in Text and Button classes 
Clearly defines which styles can be set for a widget based on context, in a human-readable way.

**Motivation**

Currently, there is no built-in way to view a widget’s set styles in a human-readable format, leading to projects like Eclipse PDE implementing complex and limited parsers for UI elements like the CSS Spy tool. This PR makes this process much simpler and easier, without requiring much upkeep. It should be quite simple to add support for and update widget classes.

**Notes**
-Since this implementation only focuses on Text and Button classes, other widget classes should be revised incrementally to use the StyleProcessor class.

-Backwards compatible; existing behavior is unchanged. 
Having both getStyles and checkStyle is a bit redundant, so merging them at some point in the future might be a good idea if it can be done without breaking existing behavior too much in downstream projects.

neilcabanilla committed:
Started working on the StyleProcessor class and the methods for it.

neilcabanilla committed:
Added print statements to the catch method in StyleProcessor.

WillPeltz committed:
Modified styleProcessor to use rule class, added thenSomeOf

WillPeltz committed:
Tentative text class implementation of styleProcessor

WillPeltz committed:
Added someOf method for non-mutually-exclusive checks, used to fix text

neilcabanilla committed:
Changed the process method in StyleProcees w/ @WillPeltz

neilcabanilla committed:
Removed the StackTrace in the catch block (process method) w/ @WillPeltz

merks commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Button.java:
```diff
@@ -278,6 +298,15 @@ static int checkStyle (int style) {
+ * @return ArrayList<String> styles
+ * @since 3.131
+ */
+public ArrayList<String> getStyles() {
```
One would never want to surface the List implementation class in any API; elsewhere too.

neilcabanilla replied:
Hello @merks,

Answering to your question...
> Given the implementation details I see, could this not all be done externally without introducing APIs in SWT?

We initially explored other approaches such as implementing it in a utility class or using annotations, however based on the comments  and guidance that we got [Issue #611](https://github.com/eclipse-pde/eclipse.pde/issues/611). We were advised to directly change the SWT repository, specifically the widget classes instead. That being said, we are flexible and happy to revise the implementation.

> One would never want to surface the List implementation class in any API; elsewhere too.

You are right with what you've said. My teammate and I are planning to change this ArrayList<String> to List<String> instead and avoiding making the method public. We've only did it this way for testing purposes to see if it works and get feedback.

Thank you so much!
Teammate - @WillPeltz

merks commented on bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/StyleProcessor.java:
```diff
@@ -0,0 +1,141 @@
+
+import org.eclipse.swt.*;
+
+public class StyleProcessor {
```
Will this really become public API or can it just be package protected?

neilcabanilla replied:
Hello merks,

You're absolutely right, we actually intended for StyleProcessor to be package private, so that it can only be accessed within the widgets package. My teammate and I forgot that part, thank you for clarifying.

merks reviewed (commented):
Given the implementation details I see, could this not all be done externally without introducing APIs in SWT?

laeubi commented:
The recommendation was to access this reflectively and not to make any public API here.

neilcabanilla committed:
Removed the public keywords & changed to package-private.

neilcabanilla committed:
Removed the comments on getStyles for Text & Button

neilcabanilla committed:
Fixed error w/ @WillPeltz of NoSuchMethod in Text class, not showing
getStyles().
# Conflicts:
#	bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Button.java
#	bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Text.java

neilcabanilla committed:
FIx error regarding, "Graphic is disposed" w/ WillPeltz
